app/controllers/pet_applications_controller.rb

```diff
@@ -8,10 +8,21 @@ def update_approval
     pet = Pet.find(params[:pet_id])
     application = Application.find(params[:application_id])
     pet.adoption_status = "pending"
-    application.approval_status = "true"
     pet.save
-    application.save
-
+    pet_application = PetApplication.find_by(pet_id: pet.id, application_id: application.id)
+    pet_application.approved = "true"
+    pet_application.save
```

Comment:
I definitely think changing the approval status to being on the pet application is a good move.